### ISSUE TITLE
When using mixins in less, the stylelint throws error which is confusing.

### ISSUE THREAD

**Issue by swastikpareek** [stylelint/stylelint#5630]
> Clearly describe the bug

The issue appears when stylelint is used with less. On a less partial file when mixin is used for a selector, the stylelint complains about the following errors

**Error**
```
- invalid variable "XXX" (less/color-no-invalid-hex)stylelint(less/color-no-invalid-hex)
- Unexpected Invalid variable  "XXX" (less/no-duplicate-variables)stylelint(less/no-duplicate-variables)
```

> Which rule, if any, is the bug related to?

`color-no-invalid-hex`, `no-duplicate-variables`

> What code is needed to reproduce the bug?

e.g.

```less
.p1 {
  color: #ccc;
}

.p2 {
  .p1();
}
```

> What Stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "extends": "stylelint-config-recommended-less",
  "plugins": ["stylelint-order"],
  "rules": {
    "at-rule-no-unknown": null,
    "selector-class-pattern": null,
    "max-nesting-depth": 3,
    "no-extra-semicolons": true,
    "no-duplicate-selectors": true,
    "at-rule-disallowed-list": ["debug"],
    "at-rule-no-vendor-prefix": true,
    "block-no-empty": true,
    "block-opening-brace-space-before": "always",
    "color-hex-case": "lower",
    "color-hex-length": "short",
    "color-named": "never",
    "color-no-invalid-hex": true,
    "declaration-bang-space-after": "never",
    "declaration-bang-space-before": "always",
    "declaration-block-semicolon-newline-after": "always",
    "declaration-block-semicolon-space-before": "never",
    "declaration-block-single-line-max-declarations": 1,
    "declaration-block-trailing-semicolon": "always",
    "declaration-colon-space-after": "always-single-line",
    "declaration-colon-space-before": "never",
    "declaration-property-value-disallowed-list": {
      "border": ["none"],
      "border-top": ["none"],
      "border-right": ["none"],
      "border-bottom": ["none"],
      "border-left": ["none"]
    },
    "function-comma-space-after": "always-single-line",
    "function-parentheses-space-inside": "never",
    "function-url-quotes": "always",
    "indentation": 2,
    "length-zero-no-unit": true,
    "media-feature-name-no-vendor-prefix": true,
    "media-feature-parentheses-space-inside": "never",
    "no-missing-end-of-source-newline": true,
    "number-leading-zero": "always",
    "number-no-trailing-zeros": true,
    "order/order": [
      [
        "custom-properties",
        "dollar-variables",
        {
          "type": "at-rule",
          "name": "extend"
        },
        {
          "type": "at-rule",
          "name": "include",
          "hasBlock": false
        },
        "declarations",
        {
          "type": "at-rule",
          "name": "include",
          "hasBlock": true
        },
        "rules"
      ]
    ],
    "order/properties-alphabetical-order": true,
    "property-no-unknown": true,
    "property-no-vendor-prefix": true,
    "rule-empty-line-before": [
      "always-multi-line",
      {
        "except": ["first-nested"],
        "ignore": ["after-comment"]
      }
    ],
    "selector-list-comma-newline-after": "always",
    "selector-max-compound-selectors": 3,
    "selector-max-id": 0,
    "selector-no-qualifying-type": true,
    "selector-no-vendor-prefix": true,
    "selector-pseudo-element-colon-notation": "double",
    "selector-pseudo-element-no-unknown": true,
    "shorthand-property-no-redundant-values": true,
    "string-quotes": "single",
    "value-no-vendor-prefix": true
  }
}
```

> Which version of Stylelint are you using?
```
"stylelint": "^13.13.1",
"stylelint-config-recommended": "^5.0.0",
"stylelint-config-recommended-less": "^1.0.1",
```

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

using CLI - `npx stylelint --config .stylelintrc <file_name>`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

e.g. "Yes, it's related to LESS mixins usage."

> What did you expect to happen?

"No warnings or errors to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following errors were flagged:"

```shell
test.css
 6:4  ✖  invalid variable "p1"               less/color-no-invalid-hex  
 6:4  ✖  Unexpected Invalid variable  "p1"   less/no-duplicate-variables
```

**Comment by wayofwade** [comment]
hello，How did you finally solve it?
